lib/ts/utils.ts

```diff
@@ -453,22 +455,68 @@ export const useOnMountAPICall = <T>(
     }
 };
 
-export function saveInvalidClaimRedirectPathInContext(userContext: any, invalidClaimRedirectPath: string) {
-    if (userContext["_default"] === undefined) {
-        userContext["_default"] = {};
-    }
-    if (userContext["_default"].redirectPath === undefined) {
-        userContext["_default"] = {
-            ...userContext["_default"],
-            invalidClaimRedirectPath,
-        };
+export const getFailureRedirectionInfo = async ({
+    invalidClaims,
+    overrideGlobalClaimValidators,
+    userContext,
+}: {
+    invalidClaims: ClaimValidationError[];
+    overrideGlobalClaimValidators?:
+        | ((globalClaimValidators: SessionClaimValidator[], userContext: any) => SessionClaimValidator[])
+        | undefined;
+    userContext: any;
+}): Promise<{ accessForbidden: boolean; redirectPath: string | undefined }> => {
+    const globalValidatorsMap = getGlobalClaimValidators({
+        overrideGlobalClaimValidators,
+        userContext,
+    }).reduce((map, validator) => {
+        map[validator.id] = validator;
+        return map;
+    }, {} as Record<string, SessionClaimValidator>);
+
+    const redirects = [];
+
+    for (const claim of invalidClaims) {
+        const validator = globalValidatorsMap[claim.validatorId];
+        if (validator.onFailureRedirection) {
+            const redirectPath = await validator.onFailureRedirection({ reason: claim.reason, userContext });
+            redirects.push(redirectPath);
+            if (redirectPath !== undefined) {
+                break;
+            }
+        }
     }
-}
+    return {
+        accessForbidden: invalidClaims.length > 0 && redirects.length === 0,
```

Comment:
Using this, if any claim onFailureRedirection returns undefined, we don't redirect to the access denied page.
This is not correct.
If:
1. The app uses a RoleClaim that redirects to the access denied page by default
2. The app later adds a claim that shouldn't redirect (so it returns undefined both redirection callbacks)

Then:
- The RoleClaim redirection no longer happens.

Reply:
The priority of redirections should be:
1. A failed claim returns a string.
2. A failed claim doesn't have `onFailureRedirection` defined -> access denied
3. A valid claim has onSuccessRedirection, and it returns a string
4. Default behaviour (success event or no redirection, depending on the location)

Reply:
- Using this, if any claim onFailureRedirection returns undefined, we don't redirect to the access denied page.
This is not correct. 
i remember we discussed that if user defines onFailureRedirection and makes it return undefined that means he does not want it to redirect to access denied